radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -184,19 +212,59 @@ public void process(
     }
   }
 
+  private void persistCurrentSubstateAccumulatorHash(Transaction dbTxn) {
+    var key = new DatabaseEntry(SUBSTATE_ACCUMULATOR_HASH_KEY);
+    this.substateAccumulatorHashDatabase.put(
+        dbTxn, key, new DatabaseEntry(this.currentSubstateAccumulatorHash));
+  }
+
+  private void logEpochHash() {
+    logger.info(
+        "Epoch Hash: {} for epoch {}. Time spent since last epoch: {} s.",
+        Bytes.toHexString(this.currentSubstateAccumulatorHash),
+        lastEpochInDbOpt,
+        watch.elapsed().toSeconds());
+  }
+
+  private OperationStatus persistEpochHash(Transaction dbTxn) {
+    return epochHashDatabase.put(
+        dbTxn,
+        new DatabaseEntry(Longs.toByteArray(lastEpochInDbOpt.orElseThrow())),
+        new DatabaseEntry(this.currentSubstateAccumulatorHash));
+  }
+
+  private void handleEpochHashAccumulatorFileUpdate() {
+    // we do not overwrite hashes and only append in a sequential way
```

Comment:
I believe the team consensus is that comments should start with a capital letter.